### PR TITLE
feat(dart): add package_config.json inventory extractor PRP: 2217

### LINKED ISSUE #2217: PRP: Extractor for Dart package_config.json
- **Software distribution method or binary type**: Dart package configuration (`.dart_tool/package_config.json`).
- **Popularity of distribution method**: Dart and Flutter use the Pub ecosystem, which contains tens of thousands of packages and serves millions of developers. The `package_config.json` file is a standard artifact in every Dart and Flutter application created since Dart 2.8. It is generated during `dart pub get` or `flutter pub get` and explicitly maps all installed dependencies to their cached file locations, capturing the exact runtime dependency tree.
- **Any critical, emergent vulnerability associated with software from the distribution method**: The OSV.dev database natively supports the Pub ecosystem. While there is no single emergent CVE, supply chain risks in widely-used Flutter packages (like `http` or `webview_flutter`) are common. SCALIBR currently extracts dependencies from the `pubspec.lock` manifest. Extracting from `package_config.json` is highly complementary, as it acts as a deployed/runtime artifact that reflects exactly which packages the Dart build system resolved and installed locally, similar to Go's `vendor/modules.txt` or PHP's `installed.json`.
- **Resources**:
  * Package config specification: https://dart.dev/go/package-config
  * Dart dependencies documentation: https://dart.dev/tools/pub/dependencies
  * OSV Pub ecosystem: https://osv.dev/list?ecosystem=Pub
  * Current supported SCALIBR inventory: https://github.com/google/osv-scalibr/blob/main/docs/supported_inventory_types.md

### PR DESCRIPTION
**Pull request by sanjay20m**
This PR adds support for extracting Dart dependencies resolved on the filesystem via `.dart_tool/package_config.json`. 

While SCALIBR currently extracts dependencies from the `pubspec.lock` manifest, extracting from `package_config.json` is highly complementary. It acts as a deployed, runtime artifact that reflects exactly which packages the Dart build system resolved and installed locally, mapping the local file paths of the cached packages.

Fixes #2217 

### Testing
Added comprehensive unit tests covering:
- Valid v1 and v2 schema parsing
- Empty and invalid JSON edge cases
- Semantic version extraction from cached path URIs
- Bypassing local directory paths (where version resolves to an empty string)

**Testbed Repository :**
[https://github.com/sanjay20m/scalibr_dart_inventory_test](https://github.com/sanjay20m/scalibr_dart_inventory_test
)
Thank you!

### CODE REVIEW

**Comment by sanjay20m** [comment]
Heyy @jess-lowe ! Please take a look at my PR and let me know if any tweaks are needed.
Thank you!

**Comment by sanjay20m** [comment]
Hey! @jess-lowe @another-rex  Please take a look at my PR and let me know if any tweaks are needed.
Thank you!

**Comment by sanjay20m** [comment]
Heyyy! @another-rex @jess-lowe  Please take a look at my PR and let me know if any tweaks are needed.

Thank you!

**Comment by sanjay20m** [comment]
Hey @another-rex, please take the time to review my PR and do tell me if any tweaks are needed.

Thank you

**Comment by sanjay20m** [comment]
Hey Reviewers! Please take the time to review my PR and let me know if any tweaks are needed.
@another-rex 
@jess-lowe 
Thank you

**Comment by sanjay20m** [comment]
Hey Reviewers! Please take the time to review my PR and let me know if any tweaks are needed.
@another-rex 
@jess-lowe 
@erikvarga 
Thank you!

**Comment by sanjay20m** [comment]
Hey Reviewers! Please take the time to review my PR and let me know if any tweaks are needed.
@another-rex 
 Thank you!

**Comment by sanjay20m** [comment]
Hey Reviewers! Please take the time to review my PR and let me know if any tweaks are needed.
@another-rex 
@jess-lowe
Thank you!

**Comment by sanjay20m** [comment]
Hey Reviewers! Please take the time to review my PR and let me know if any tweaks are needed.

@another-rex 

@Ly-Joey

@jess-lowe

Thanks!

**Comment by sanjay20m** [comment]
Hey reviewers please take a look at my PR.

@another-rex

@Ly-Joey

@jess-lowe

Thanks!

**Comment by sanjay20m** [comment]
Hey @another-rex  I wanted to ask about #2295. It has been waiting for review for quite some time, and I’m not sure if there is something wrong with the PR or if there’s any issue preventing it from being reviewed.

The checks are passing, and I’ve addressed the changes that came up during development. Could you please let me know if there’s anything currently blocking the review, or if there’s something I need to fix on my side?

I’d really appreciate knowing what’s holding the review up so I can address it.

Thanks!

**Comment by G-Rath** [comment]
@sanjay20m the current blocker is "time" - the team have a lot of different things on their plate between scalibr, scanner, and dev.

There's a PRP meeting at least once a month and the team are slowly working their way through the open PRs, so your contribution will be reviewed eventually. In the meantime repeatedly `@`ing the team won't help and just creates more noise for them.

fwiw while I'm not on the PRP team, I have had a look over your PR and nothing has jumped out as surprising so I would expect this is mainly a matter of the team having the bandwidth to do the review.

File: extractor/filesystem/list/list.go [hunk 1]
```diff
@@ -281,7 +282,7 @@ var (
 		gobinary.Name: {protoCfg(gobinary.New)},
 	}
 	// DartSource extractors for Dart.
-	DartSource = InitMap{pubspec.Name: {protoCfg(pubspec.New)}}
+	DartSource = InitMap{packageconfig.Name: {protoCfg(packageconfig.New)}, pubspec.Name: {protoCfg(pubspec.New)}}
```
Comment (G-Rath) [inline extractor/filesystem/list/list.go]: nit: we typically make these multi-line when there's more than one

**Comment by sanjay20m** [comment]
Thanks for the context, @G-Rath that makes sense, and I really appreciate you taking the time to look it over and let me know nothing jumped out. Good to know it's just a bandwidth thing on the team's side. I'll be patient and wait for the PRP review cycle rather than pinging again. Thanks again for the update!

**Review by Ly-Joey — APPROVED** [review]
Hi @sanjay20m, sorry for the wait while we work through the PRP backlog. This change LGTM!

**Comment by Ly-Joey** [comment]
Hi @sanjay20m, could you please also add some E2E test to osv-scanner? Thanks!

**Comment by sanjay20m** [comment]
> Hi @sanjay20m, could you please also add some E2E test to osv-scanner? Thanks!

Hey @Ly-Joey , I just opened the PR over in `osv-scanner` and added the tests like you asked! Thanks !!
